packages/regl-worldview/src/WorldviewContext.js

```diff
@@ -227,7 +227,22 @@ export class WorldviewContext {
     this.counters.render = Date.now() - start;
   }
 
-  _debouncedPaint = debounce(this.paint, 10);
+  /**
+   * Be careful to not use debounce which can delay paint indefinitely.
+   * Also be careful not to use something like Lodash throttle (which uses debounce)
+   * which as of 3/26/2020 appears to be erroneously using setTimeout instead of
+   * requestAnimationFrame. We want the latter so that paint calls can be sync'd
+   * with the browser's animation loop for maximum performance.
+   */
+  _throttledPaint = () => {
+    const now = performance.now();
+    if (now - this._lastPaintCall < 10) {
+      console.warn("Worldview paint has been throttled");
```

Comment:
Might be good to not log this warning since it might get a bit spammy for heavy applications, or maybe do it conditionally using `process.env.NODE_ENV !== "production"` or something. Not sure how we do that in other places.

Reply:
I had considered this, but I leaned towards logging a warning so a consumer can know they are doing something wrong, which is arguably better than not knowing.

As a consumer I'd rather get a warning than have my app mysteriously not update (due to the debounce).

Maybe we don't need it, but then consumers could just cargo cult their usage of the paint call, calling it all over the place when it isn't needed. I guess that would be fine now that it uses throttle instead of debounce, but feels less ergonomic than with the warning, which was inspired by a similar warning in one of our internal repos. Ideally we'd surface to the consumer's developers that they are doing something wrong (calling paint out of sync), which causes a perf issue.

Reply:
+1.

Reply:
Ahh I see, yeah I don't know this code well enough to realise that that was actually going to be incorrect behaviour. In that case it might be good to include some actionable information in the warning message to let the developer know what they should do to fix the issue.